platforms/documentation/docs/src/snippets/how-to/share-outputs/groovy/producer/build.gradle

```
@@ -12,17 +16,23 @@ def instrumentedJar = tasks.register("instrumentedJar", Jar) {
 
 // tag::custom-config[]
 configurations {
+    // Create a custom consumable configuration named 'instrumentedJars'
+    // This allows other projects to depend on the instrumented JAR variant
     create("instrumentedJars") {
         canBeConsumed = true
         canBeResolved = false
+
+        // Assign attributes so that consuming projects can match on these
         attributes {
             attribute(Category.CATEGORY_ATTRIBUTE, objects.named(Category, Category.LIBRARY))
             attribute(Usage.USAGE_ATTRIBUTE, objects.named(Usage, Usage.JAVA_RUNTIME))
+            // The defining attribute
             attribute(LibraryElements.LIBRARY_ELEMENTS_ATTRIBUTE, objects.named(LibraryElements, "instrumented-jar"))
         }
     }
 }
 
+// Publish the custom JAR artifact to the 'instrumentedJars' configuration
```

```suggestion
// Add the custom JAR artifact to the 'instrumentedJars' configuration
```

I'd like to avoid overloading the "Publish" term. IMO publishing implies actually going and uploading files to a repository.

---

Done.